packages/firestore/src/local/indexeddb_remote_document_cache.ts

```diff
@@ -331,95 +295,62 @@ export class IndexedDbRemoteDocumentCache implements RemoteDocumentCache {
   getNewDocumentChanges(
     transaction: PersistenceTransaction
   ): PersistencePromise<MaybeDocumentMap> {
-    assert(
-      this.keepDocumentChangeLog,
-      'Can only call getNewDocumentChanges() when document change log is enabled'
-    );
-    let changedKeys = documentKeySet();
     let changedDocs = maybeDocumentMap();
 
-    const range = IDBKeyRange.lowerBound(
-      this._lastProcessedDocumentChangeId + 1
+    const lastReadTime = this.serializer.toDbTimestampKey(
+      this.lastProcessedReadTime
     );
-    let firstIteration = true;
-
-    const changesStore = documentChangesStore(transaction);
-    return changesStore
-      .iterate({ range }, (_, documentChange) => {
-        if (firstIteration) {
-          firstIteration = false;
-
-          // If our client was throttled for more than 30 minutes, another
-          // client may have garbage collected the remote document changelog.
-          if (this._lastProcessedDocumentChangeId + 1 !== documentChange.id) {
-            // Reset the `lastProcessedDocumentChangeId` to allow further
-            // invocations to successfully return the changes after this
-            // rejection.
-            return this.synchronizeLastDocumentChangeId(changesStore).next(() =>
-              PersistencePromise.reject(
-                new FirestoreError(
-                  Code.DATA_LOSS,
-                  REMOTE_DOCUMENT_CHANGE_MISSING_ERR_MSG
-                )
-              )
-            );
-          }
-        }
 
-        changedKeys = changedKeys.unionWith(
-          this.serializer.fromDbResourcePaths(documentChange.changes)
-        );
-        this._lastProcessedDocumentChangeId = documentChange.id!;
-      })
-      .next(() => {
-        const documentPromises: Array<PersistencePromise<void>> = [];
-        changedKeys.forEach(key => {
-          documentPromises.push(
-            this.getEntry(transaction, key).next(maybeDocument => {
-              const doc =
-                maybeDocument ||
-                new NoDocument(key, SnapshotVersion.forDeletedDoc());
-              changedDocs = changedDocs.insert(key, doc);
-            })
+    const documentsStore = remoteDocumentsStore(transaction);
+    const range = IDBKeyRange.lowerBound(lastReadTime, true);
+    return documentsStore
+      .iterate(
+        { index: DbRemoteDocument.readTimeIndex, range },
+        (_, dbRemoteDoc) => {
+          const doc = this.serializer.fromDbRemoteDocument(dbRemoteDoc);
```

Comment:
Maybe add a comment that we explicitly don't use maybeDecodeDocument() because we want to keep sentinel deletes?